pkg/schema/converter/discovery_test.go

```diff
@@ -0,0 +1,419 @@
+							"group":      "TestGroup",
+							"version":    "v1",
+							"kind":       "TestResource",
+							"resource":   "testResources",
+							"verbs":      []string{"get"},
+							"namespaced": true,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "discovery error but still got some information",
+			discoveryErr: &discovery.ErrGroupDiscoveryFailed{Groups: map[schema.GroupVersion]error{
+				schema.GroupVersion{
+					Group:   "NotFound",
+					Version: "v1",
+				}: fmt.Errorf("group Not found"),
+			},
+			},
+			groups: []schema.GroupVersion{{Group: "TestGroup", Version: "v1"}},
+			resources: map[schema.GroupVersion][]metav1.APIResource{
+				{Group: "TestGroup", Version: "v1"}: {
+					{
+
+						Name:         "testResources",
+						SingularName: "testResource",
+						Kind:         "TestResource",
+						Namespaced:   true,
+						Verbs:        metav1.Verbs{"get"},
+					},
+				},
+			},
+			wantError: false,
+			desiredSchema: map[string]*types.APISchema{
+				"testgroup.v1.testresource": {
+					Schema: &wranglerSchema.Schema{
+						ID:         "testgroup.v1.testresource",
+						PluralName: "TestGroup.v1.testResources",
+						Attributes: map[string]interface{}{
+							"group":      "TestGroup",
+							"version":    "v1",
+							"kind":       "TestResource",
+							"resource":   "testResources",
+							"verbs":      []string{"get"},
+							"namespaced": true,
+						},
+					},
+				},
+			},
+		},
+		{
+			name:         "discovery error, not partial",
+			discoveryErr: fmt.Errorf("cluster unavailable"),
+			groups:       []schema.GroupVersion{{Group: "TestGroup", Version: "v1"}},
+			resources: map[schema.GroupVersion][]metav1.APIResource{
+				{Group: "TestGroup", Version: "v1"}: {
+					{
+						Name:         "testResources",
+						SingularName: "testResource",
+						Kind:         "TestResource",
+						Namespaced:   true,
+						Verbs:        metav1.Verbs{"get"},
+					},
+				},
+			},
+			wantError:     true,
+			desiredSchema: map[string]*types.APISchema{},
+		},
+		{
+			name:   "bad group version",
+			groups: []schema.GroupVersion{{Group: "Invalid/Group", Version: "v2"}},
+			resources: map[schema.GroupVersion][]metav1.APIResource{
+				{Group: "Invalid/Group", Version: "v2"}: {
+					{
+						Name:         "testResources",
+						SingularName: "testResource",
+						Kind:         "TestResource",
+						Namespaced:   true,
+						Verbs:        metav1.Verbs{"get"},
+					},
+				},
+			},
+			wantError: true,
+			desiredSchema: map[string]*types.APISchema{
+				"core..testresource": {
+					Schema: &wranglerSchema.Schema{
+						ID:         "core..testresource",
+						PluralName: "core..testResources",
+						Attributes: map[string]interface{}{
+							"group":      "",
+							"version":    "",
+							"kind":       "TestResource",
+							"resource":   "testResources",
+							"verbs":      []string{"get"},
+							"namespaced": true,
+						},
+					},
+				},
+			},
+		},
```

Comment:
That would be how I read the code - it looks like even if it can't parse the group version it [still tries to refresh](https://github.com/rancher/steve/blob/9935404019063126b6ac856db1ad56e0bc457ff8/pkg/schema/converter/discovery.go#L37-L42) without continuing. The test also reads that way to me.